server/src/utils/auth.ts

```diff
@@ -0,0 +1,46 @@
+import axios, { type AxiosError } from 'axios';
+
+export type UserInfo = {
+  email: string;
+  email_verified: string;
+  name: string;
+  nickname: string;
+  picture: string;
+  sub: string;
+  updated_at: string;
+};
+
+export type UserInfoResponse =
+  | {
+      success: true;
+      userInfo: UserInfo;
+    }
+  | {
+      success: false;
+      reason: 'AXIOS_ERROR';
+      error: AxiosError;
```

Comment:
Don't make this specific to axios. We're basically exposing what library we're using for API calls. This should be an error that we define. Maybe `AUTH_ERROR`?